cmd/axiom/auth/auth_login.go

```diff
@@ -67,75 +62,105 @@ func newLoginCmd(f *cmdutil.Factory) *cobra.Command {
 			$ axiom auth login
 			
 			# Provide parameters on the command-line:
-			$ echo $AXIOM_ACCESS_TOKEN | axiom auth login --url="https://axiom.eu-west-1.aws.com" --alias="axiom-eu-west-1" --token-stdin --token-type="personal" -f
+			$ echo $AXIOM_ACCESS_TOKEN | axiom auth login --alias="axiom-eu-west-1" --url="https://axiom.eu-west-1.aws.com" -f
 		`),
 
-		PreRunE: func(*cobra.Command, []string) error {
-			if !opts.IO.IsStdinTTY() && opts.TokenStdIn {
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			if !opts.IO.IsStdinTTY() {
 				return nil
 			}
-			return completeLogin(opts)
+			return completeLogin(cmd.Context(), opts)
 		},
 
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if opts.TokenType != config.Personal && opts.TokenType != config.Ingest {
-				return fmt.Errorf("unknown token type %q (choose %q or %q)",
-					opts.TokenType, config.Personal, config.Ingest)
-			}
 			return runLogin(cmd.Context(), opts)
 		},
 	}
 
 	cmd.Flags().StringVar(&opts.URL, "url", "", "Url of the deployment")
 	cmd.Flags().StringVarP(&opts.Alias, "alias", "a", "", "Alias of the deployment")
-	cmd.Flags().BoolVar(&opts.TokenStdIn, "token-stdin", false, "Read token from stdin")
-	cmd.Flags().StringVarP(&opts.TokenType, "token-type", "t", "", "Type of the token (choose \"personal\" or \"ingest\")")
```

Comment:
This breaks existing scripts, maybe it's better to deprecate? Otoh we're still at 0.x so breaking are kind of expected.

Reply:
Let's not pollute pre-v1.0.0 with deprecations. We have properly tagged releases so at least nothing breaks when not updating. The upgrade process also is rather simple: Just drop the flags.

But thanks for the comment, found some dead code not catched by any linters and removed some duplicated logic as well.